x-pack/plugins/cases/server/features.ts

```diff
@@ -20,44 +21,81 @@ import { APP_ID, FEATURE_ID } from '../common/constants';
 
 const FEATURE_ORDER = 3100;
 
-export const getCasesKibanaFeature = (): KibanaFeatureConfig => ({
-  id: FEATURE_ID,
-  name: i18n.translate('xpack.cases.features.casesFeatureName', {
-    defaultMessage: 'Cases',
-  }),
-  category: DEFAULT_APP_CATEGORIES.management,
-  app: [],
-  order: FEATURE_ORDER,
-  management: {
-    insightsAndAlerting: [APP_ID],
-  },
-  cases: [APP_ID],
-  privileges: {
-    all: {
-      cases: {
-        all: [APP_ID],
-      },
-      management: {
-        insightsAndAlerting: [APP_ID],
-      },
-      savedObject: {
-        all: [],
-        read: [],
-      },
-      ui: ['crud_cases', 'read_cases'],
+export const getCasesKibanaFeature = (): KibanaFeatureConfig => {
+  const capabilities = createUICapabilities();
+
+  return {
+    id: FEATURE_ID,
+    name: i18n.translate('xpack.cases.features.casesFeatureName', {
+      defaultMessage: 'Cases',
+    }),
+    category: DEFAULT_APP_CATEGORIES.management,
+    app: [],
+    order: FEATURE_ORDER,
+    management: {
+      insightsAndAlerting: [APP_ID],
     },
-    read: {
-      cases: {
-        read: [APP_ID],
-      },
-      management: {
-        insightsAndAlerting: [APP_ID],
+    cases: [APP_ID],
+    privileges: {
+      all: {
+        cases: {
+          create: [APP_ID],
+          read: [APP_ID],
+          update: [APP_ID],
+          push: [APP_ID],
+        },
+        management: {
+          insightsAndAlerting: [APP_ID],
+        },
+        savedObject: {
+          all: [],
+          read: [],
+        },
+        ui: capabilities.all,
       },
-      savedObject: {
-        all: [],
-        read: [],
+      read: {
+        cases: {
+          read: [APP_ID],
+        },
+        management: {
+          insightsAndAlerting: [APP_ID],
+        },
+        savedObject: {
+          all: [],
+          read: [],
+        },
+        ui: capabilities.read,
       },
-      ui: ['read_cases'],
     },
-  },
-});
+    subFeatures: [
+      {
+        name: i18n.translate('xpack.cases.features.deleteSubFeatureName', {
+          defaultMessage: 'Delete',
+        }),
+        privilegeGroups: [
+          {
+            groupType: 'independent',
+            privileges: [
+              {
+                api: [],
+                id: 'cases_delete',
+                name: i18n.translate('xpack.cases.features.deleteSubFeatureDetails', {
+                  defaultMessage: 'Delete entities',
```

Comment:
@nastasha-solomon and @lcawl any thoughts for what we could use as the description for what a user is able to delete once they are granted delete permissions? Basically we want to communicate that when you have delete you can delete everything (cases themselves, comments, etc). Maybe `Delete case entities`?

Reply:
+1 to `Delete cases and comments`. Unless there is any other entity which might be added to the list in the short to mid-term?

Reply:
@jonathan-buttner how about `Delete cases and case data` or `Delete cases and related case data`? This encompasses case comments, case/user history, and various case settings. If you wanted to elaborate on the scope of the privs or provide users with more information about the setting, maybe you could add an informational icon after `Delete` ? Here's an example of the icon being used under Management priv settings. I'm not sure if there's guidelines or standards for using the icon in this flyout, so feel free to ignore my recommendation if it goes against standards. 
<img width="766" alt="ui-text" src="https://user-images.githubusercontent.com/79124755/178285214-22769cab-3783-447a-a960-fe5444a46f4e.png">

Reply:
Hm, looks like the informational icon is mainly used to convey pre-reqs or related priv requirements. I also don't see it used anywhere else other than the Management section, so maybe it's something that's used sparingly, sort of like a last resort? Looking at these examples, it might not be suitable to use in the way I described.
<img width="746" alt="informational-icon" src="https://user-images.githubusercontent.com/79124755/178287699-5a47c613-d4f8-46ef-995e-8728c4a65256.png">

Reply:
If we go with `Delete cases and comments` we would need to remember to update it if we add new entities. On the other hand, it is more descriptive. I lean towards `Delete cases and case data` as it is a bit more generic. @peteharverson What do you think?